### PR TITLE
Some refactoring suggestions

### PR DESCRIPTION
**Pull request by mayorova**
It was hard to explain in the PR review... I thought that we could improve some variables naming, to make clearer what they represent.

### CODE REVIEW

File: app/javascript/src/AccessTokens/components/ExpirationDatePicker.tsx [hunk 1]
```diff
@@ -32,46 +32,51 @@ interface Props {
 }
 
 const ExpirationDatePicker: FunctionComponent<Props> = ({ id, label, tzOffset }) => {
-  const [selectedItem, setSelectedItem] = useState(collection[0])
-  const [pickedDate, setPickedDate] = useState(new Date())
+  const [dropdownSelectedItem, setDropdownSelectedItem] = useState(collection[0])
```
Comment (mayorova) [inline app/javascript/src/AccessTokens/components/ExpirationDatePicker.tsx:35]: To make it more cleare, that this is for the item in the dropbox.

File: app/javascript/src/AccessTokens/components/ExpirationDatePicker.tsx [hunk 2]
```diff
@@ -32,46 +32,51 @@ interface Props {
...
+  const [calendarPickedDate, setCalendarPickedDate] = useState(new Date())
```
Comment (mayorova) [inline app/javascript/src/AccessTokens/components/ExpirationDatePicker.tsx:36]: Make it clear that this is for calendar.

File: app/javascript/src/AccessTokens/components/ExpirationDatePicker.tsx [hunk 3]
```diff
@@ -32,46 +32,51 @@ interface Props {
...
-    let value = ''
+  const selectedDate= useMemo(() => {
+    let value = null
 
-    if (fieldDate) {
-      value = fieldDate.toISOString()
-    } else if (selectedItem.id === 'custom' ) {
-      value = pickedDate.toISOString()
+    if (dropdownDate) {
+      value = dropdownDate
```
Comment (mayorova) [inline app/javascript/src/AccessTokens/components/ExpirationDatePicker.tsx:50]: I didn't like the fact that we first convert the date to string, and then in `formattedDateValue` we pparsed it again into Date.
The idea is that "the source of truth" for the selected date is a `Date` instance, and then it is formatted for the representation purposes as the last step.

File: app/javascript/src/AccessTokens/components/ExpirationDatePicker.tsx [hunk 4]
```diff
@@ -32,46 +32,51 @@ interface Props {
...
+    return formatter.format(selectedDate)
+  }, [selectedDate])
 
   const fieldHint = useMemo(() => {
     if (!formattedDateValue) return
 
     return `The token will expire on ${formattedDateValue}`
   }, [formattedDateValue])
 
+  const inputDateValue = useMemo(() => {
```
Comment (mayorova) [inline app/javascript/src/AccessTokens/components/ExpirationDatePicker.tsx:74]: Named like this so it's clear that this is for the input value.

Also, it seems that now variables ending with `Date` are actually Dates (`selectedDate`, `calendarPickedDate`, `dropdownDate`), and the ones ending with `DateValue` are string representaions (`inputDatevalue`, `formattedDateValue`).
This was not on purpose, but I think it makes sense :grimacing:

File: app/javascript/src/AccessTokens/components/ExpirationDatePicker.tsx [hunk 5]
```diff
@@ -144,11 +149,11 @@ const ExpirationDatePicker: FunctionComponent<Props> = ({ id, label, tzOffset })
...
       </FormGroup>
-      <input id={id} name={id} type="hidden" value={dateValue} />
-      {selectedItem.id === 'custom' && (
-        <CalendarMonth className="pf-u-mt-md" date={pickedDate} onChange={setPickedDate} />
+      <input id={id} name={id} type="hidden" value={inputDateValue} />
+      {dropdownSelectedItem.id === 'custom' && (
+        <CalendarMonth className="pf-u-mt-md" date={calendarPickedDate} onChange={setCalendarPickedDate} />
       )}
-      {dateValue === '' && (
+      {!inputDateValue && (
```
Comment (mayorova) [inline app/javascript/src/AccessTokens/components/ExpirationDatePicker.tsx:156]:
```suggestion
      {!selectedDate && (
```

**Comment by jlledom** [comment]
I applied the suggestions

**Pull request closed without merging**